modules/basics/src/main/java/com/opengamma/basics/date/DayCount.java

```diff
@@ -79,15 +80,52 @@ public default double yearFraction(LocalDate firstDate, LocalDate secondDate) {
    * Gets the year fraction between the specified dates.
    * <p>
    * Given two dates, this method returns the fraction of a year between these
-   * dates according to the convention.
+   * dates according to the convention. The dates must be in order.
    * 
    * @param firstDate  the first date
    * @param secondDate  the second date, on or after the first date
    * @param scheduleInfo  the schedule information
-   * @return the year fraction
+   * @return the year fraction, zero or greater
+   * @throws IllegalArgumentException if the dates are not in order
+   * @throws UnsupportedOperationException if the year fraction cannot be obtained
+   */
+  public abstract double yearFraction(LocalDate firstDate, LocalDate secondDate, ScheduleInfo scheduleInfo);
+
+  /**
+   * Gets the relative year fraction between the specified dates.
+   * <p>
+   * Given two dates, this method returns the fraction of a year between these
+   * dates according to the convention. If the first date is after the second date,
+   * the result is calculated by swapping the dates to be in order and negating the result.
```

Comment:
This sentence seems to be about potential implementations not the API. You have said it calculates between the 2 dates - you don't really need to say anything more (i.e. "numbers between 10 & 4" makes as much sense as "numbers between 4 & 10"). And what it does say doesn't actually explain what it does anyway.